### PR TITLE
Add isolates() function to rustworkx

### PR DESCRIPTION
**Pull request by mtreinish**
This commit adds a new function, isolates(), which is used to find all the isolates in a graph.

### CODE REVIEW

File: rustworkx-core/src/connectivity/isolates.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use petgraph::visit::{IntoNeighborsDirected, IntoNodeIdentifiers, NodeIndexable};
+use petgraph::Direction::{Incoming, Outgoing};
+
+/// Return the fisolates in a graph object
```
Comment (prakharb10) [inline rustworkx-core/src/connectivity/isolates.rs:16]: I think there's a small typo here 😅

Comment (mtreinish) [reply rustworkx-core/src/connectivity/isolates.rs:16]: Ah good catch, thanks
```suggestion
/// Return the isolates in a graph object
```

**Review by IvanIsCoding — APPROVED** [review]
LGTM

**Review by enavarro51 — APPROVED** [review]
This LGTM. Thanks for the addition.

**Pull request merged**